File: bro-agent-api/src/main/python/bro_agent/agent/RestoreExecutionState.py
import logging
import bro_agent.agent.states

from bro_agent.agent.RestoreExecutionActions import RestoreExecutionActions
from bro_agent.generated.Action_pb2 import Action
from bro_agent.generated.OrchestratorControl_pb2 import OrchestratorMessageType


class RestoreExecutionState(bro_agent.agent.RestoreState.RestoreState):
    """ Performs restore """

    def __init__(self, agent, restore_information):
        # @param agent The agent performing the restore
        # @param restoreInformation information required for restore
        bro_agent.agent.RestoreState.RestoreState.__init__(self, agent, restore_information)

    def process_message(self, message):
        if self.is_post_actions_message(message):
            logging.info("Received restore post action message for backup: %s", self.restore_information.backup_name)
            return bro_agent.agent.RestorePostActionState.RestorePostActionState(self.agent, self.restore_information)
        elif self.is_cancel_message(message):
            logging.info("Received cancel restore message for backup: %s", self.restore_information.backup_name)
            return self.cancel_action()

        self.agent.send_stage_complete_message(False, "Unsuccessful restore stage", Action.RESTORE)
        return bro_agent.agent.WaitingForActionState.WaitingForActionState(self.agent)

    @staticmethod
    def is_post_actions_message(message):
        """ Is the message a postaction? """
        return (message.orchestratorMessageType == OrchestratorMessageType.POST_ACTIONS and message.postActions is not None)

    def trigger(self):
        self.agent.execute_restore(RestoreExecutionActions(self.agent, self.restore_information))

    def finish_action(self):
        self.agent.send_stage_complete_message(False,
                                               "Finish action called during restore execution",
                                               Action.RESTORE)
        return bro_agent.agent.FinishedActionState.FinishedActionState(self.agent,
                                                                       self.restore_information.backup_name,
                                                                       Action.RESTORE)

File: bro-agent-api/src/main/java/com/ericsson/adp/mgmt/bro/api/agent/RestoreExecutionState.java
/*------------------------------------------------------------------------------
 *******************************************************************************
 * COPYRIGHT Ericsson 2019
 *
 * The copyright to the computer program(s) herein is the property of
 * Ericsson Inc. The programs may be used and/or copied only with written
 * permission from Ericsson Inc. or in accordance with the terms and
 * conditions stipulated in the agreement/contract under which the
 * program(s) have been supplied.
 *******************************************************************************
 *----------------------------------------------------------------------------*/
package com.ericsson.adp.mgmt.bro.api.agent;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import com.ericsson.adp.mgmt.action.Action;
import com.ericsson.adp.mgmt.control.OrchestratorControl;
import com.ericsson.adp.mgmt.control.OrchestratorMessageType;

/**
 * Performs restore
 */
public class RestoreExecutionState extends RestoreState {

    private static final Logger log = LogManager.getLogger(RestoreExecutionState.class);

    /**
     * @param agent The agent performing the restore
     * @param restoreInformation information required for restore
     */
    public RestoreExecutionState(final Agent agent, final RestoreInformation restoreInformation) {
        super(agent, restoreInformation);
    }

    @Override
    public AgentState processMessage(final OrchestratorControl message) {
        if (isPostActionsMessage(message)) {
            log.info("Received restore post action message for backup: {}", this.restoreInformation.getBackupName());
            return new RestorePostActionState(this.agent, this.restoreInformation);
        } else if (isCancelMessage(message)) {
            log.info("Received cancel restore message for backup: {}", this.restoreInformation.getBackupName());
            return cancelAction();
        }
        this.agent.sendStageCompleteMessage(false, "Unsuccessful restore stage", Action.RESTORE);
        return new WaitingForActionState(this.agent);
    }

    private boolean isPostActionsMessage(final OrchestratorControl message) {
        return message.getOrchestratorMessageType().equals(OrchestratorMessageType.POST_ACTIONS) && message.hasPostActions();
    }

    @Override
    public void trigger() {
        this.agent.executeRestore(new RestoreExecutionActions(this.agent, this.restoreInformation));
    }

    @Override
    public AgentState finishAction() {
        this.agent.sendStageCompleteMessage(false, "Finish action called during restore execution", Action.RESTORE);
        return new FinishedActionState(this.agent, this.restoreInformation.getBackupName(), Action.RESTORE);
    }

}
